core/src/mindustry/ui/dialogs/JoinDialog.java

```diff
@@ -563,25 +563,32 @@ private void saveServers(){
         transient Host lastHost;
 
         void setIP(String ip){
-
-            //parse ip:port, if unsuccessful, use default values
-            if(ip.lastIndexOf(':') != -1 && ip.lastIndexOf(':') != ip.length() - 1){
-                try{
+            try{
+                boolean isIpv6 = ip.chars().filter(c -> c == ':').count() > 1;
```

Comment:
`chars()` is a Java 8 feature. Don't use it.

Reply:
shit right sorry

Reply:
Note: `Strings#count(String, char)`

Reply:
done